### PR TITLE
lora: sx1272: Add SX1272 LoRa radio support

### PR DESCRIPTION
**Pull request by abxy**
~NOTE!!! This PR depends on https://github.com/zephyrproject-rtos/loramac-node/pull/6 which is not yet merged.~

sx1272.c created as a copy of sx1276.c and
the replacement 1276 -> 1272

Signed-off-by: Alexander Mihajlovic <a@abxy.se>

### CODE REVIEW

File: drivers/lora/sx1272.c [hunk 1]
```diff
@@ -0,0 +1,567 @@
...
+#define SX1272_PA_OUTPUT(power)				\
+	DT_ENUM_IDX(DT_DRV_INST(0), power_amplifier_output)
+#else
+BUILD_ASSERT(0, "None of rfo-enable-gpios, pa-boost-enable-gpios and "
+	     "power-amplifier-output has been specified. "
+	     "Look at semtech,sx1272.yaml to fix that.");
+#endif
+
+#define SX1272_REG_PA_CONFIG			0x09
+#define SX1272_REG_PA_DAC			0x4d
```
Comment (mniestroj) [inline drivers/lora/sx1272.c:69]: This should be `0x5a` according to DS.

**Review by mniestroj — CHANGES_REQUESTED** [review]
We should definitely have one common driver for sx1276 and sx1272, there is too much stuff copy pasted with `s/sx1276/sx1272/` right now. There are minor differences in register map of both radios, but they should be addressed within single source file using ifdefs (similar to how lis2dh sensor driver supports multiple models).

**Comment by andysan** [comment]
I definitely agree with @mniestroj's observation that there should be a common driver (or at least more shared code) between the sx1276 and sx1272 drivers. The same observation goes for the device tree specification since there is not reason to duplicate that if most of the driver is shared (see for example `dts/bindings/lora/semtech,sx126*.yaml`).

**Comment by mbolivar-nordic** [comment]
DT binding looks reasonable; I'm not qualified to review the rest.

**Comment by Mani-Sadhasivam** [comment]
> I definitely agree with @mniestroj's observation that there should be a common driver (or at least more shared code) between the sx1276 and sx1272 drivers. The same observation goes for the device tree specification since there is not reason to duplicate that if most of the driver is shared (see for example `dts/bindings/lora/semtech,sx126*.yaml`).

Yes I agree too. @abxy Please split out the common parts of the driver to avoid duplication and do the same for binding too!

File: dts/bindings/lora/semtech,sx1272.yaml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+include: spi-device.yaml
+
+properties:
+    reset-gpios:
+      type: phandle-array
+      required: true
+      description: |
+        GPIO connected to the sensor NRST signal.
+
+        This signal is open-drain, active-low as interpreted by the
```
Comment (tagunil) [inline dts/bindings/lora/semtech,sx1272.yaml:18]: This is not right. SX1272/3 have active-high RESET signal.

**Comment by abxy** [comment]
Closing this PR because https://github.com/zephyrproject-rtos/zephyr/pull/31576 does it better.

**Pull request closed without merging**